content/en/blog/_posts/2023-08-04-kubernetes-1.28-blog.md

```diff
@@ -0,0 +1,299 @@
+there pods will be deleted by its kubelet and new pods can be created on a different running node.
+If the original node does not come up (common with an [immutable infrastructure](https://glossary.cncf.io/immutable-infrastructure/) design),
+those pods would be stuck in a `Terminating` status on the shut-down node forever.
+
+This handles node shutdown cases that are not detected by Kubelet's Node Shutdown Manager. The pods will be forcefully deleted in this case, triggering the deletion of the VolumeAttachments, and new pods will be created on a new running node so that application can continue to function.
+    
+## Improvements to CustomResourceDefinition validation expressions 
+
+The [Common Expression Language (CEL)](https://github.com/google/cel-go) can be used to validate
+[custom resources](https://kubernetes.io/docs/concepts/extend-kubernetes/api-extension/custom-resources/). The primary goal is to allow the majority of the validation use cases that might once have needed you, as a
+CustomResourceDefinition (CRD) author, to design and implement a webhook.
+Instead, and as a beta feature, you can add _validation expressions_ directly into the schema of a CRD.
+    
+CRDs need direct support for non-trivial validation. While admission webhooks do support CRDs validation, they significantly complicate the development and operability of CRDs.
+
+This KEP proposes that an inline expression language be integrated directly into CRDs such that a much larger portion of validation use cases can be solved without the use of webhooks. When selecting an expression language, we want to be sure that it can support defaulting and CRD conversion in the future.
+
+It is sufficiently lightweight and safe to be run directly in the kube-apiserver (since CRD creation is a privileged operation), has a straight-forward and unsurprising grammar, and supports pre-parsing and type checking of expressions, allowing syntax and type errors to be caught at CRD registration time.
+
+## ValidatingAdmissionPolicies graduate to beta
+    
+Common Expression language for admission control is customizable, in-process validation of requests to the Kubernetes API server as an alternative to validating admission webhooks.
+
+This builds on the capabilities of the CRD Validation Rules feature that graduated to beta in 1.25 but with a focus on the policy enforcement capabilities of validating admission control.
+    
+This will lower the infrastructure barrier to enforcing customizable policies as well as providing primitives that help the community establish and adhere to the best practices of both K8s and its extensions.
+
+## CRD validation ratcheting
+    
+This Allows custom resources to fail validation if the patch did not alter any of the invalid fields.  The ability to shift left validation logic from controllers to the front end is a long-term goal for improving the useability of the Kubernetes project.
```

Comment:
> The ability to shift left validation logic from controllers to the front end is a long-term goal for improving the useability of the Kubernetes project.

For CRD Validation Ratcheting specifically, I'm not sure this is accurate statement in isolation. CRD Validation Ratcheting was added in part to support the Declarative Validation effort which does have an effect of shifting validation left, ratcheting errors for CRDs does not itself do this.

I'd say for K8S users the benefits are:

1. Improved UX knowing that previously applied resources will remain applied
2. CRDs can change over time to allow for most validations to be made more strict without requiring a new version and conversion webhook.

Reply:
Hi we have decided not to add this feature to the blog in the end after discussion, we didn't find a way to present it to readers without confusing them. Thanks for your help we appreciate it.